### PR TITLE
Push __Test__ folder to main

### PR DESCRIPTION
**Pull request by cfiguer055**
The `__test__` branch contains a suite of tests for verifying the functionality and stability of the Smart Clothing App. You can find a detailed test report here:
[SmartClothingApp-TestReport.pdf](https://github.com/user-attachments/files/17336575/SmartClothingApp-TestReport.pdf)

### Test Coverage Overview

1. **Integration Tests:**
   - **Auth Stack Tests:** Focus on testing authentication-related functionality (e.g., SignInHooks, SignUpScreen).
   - **Component and Screen Navigation:** Verify navigation flow from Sign-In and Sign-Up screens to other areas of the app (e.g., SigninToHome, SignupToHome).
   - **Main Tab Navigation:** Validate transitions between main tabs like Home, Profile, and Settings.
   - **Main Screen Tests:** Cover key screens like `ViewInsightsScreen` and `SettingsScreen`, ensuring the correct functionality and data rendering.
   - **User Actions:** Check integration of user actions within the app.

2. **Unit Tests:**
   - **Actions and Reducers:** Focus on core application logic, such as app actions and reducers (e.g., `userActions`, `toastReducer`).
   - **Components:** Test individual components like `AppHeader` to ensure correct behavior in isolation.
   - **Utils:** Cover utility functions (e.g., `localStorage`).

3. **Snapshots:** Snapshots help ensure the UI remains consistent, with automated checks to detect unintended changes.

### Additional Structure:
- **Mocks:** Provide mock implementations of third-party libraries and dependencies for isolated testing.
- **Logs and Archives:** Store test execution logs and archive older tests to maintain version control and test history.

---

### GitHub Issues Summary
This pull request addresses several key testing issues, with some completed and others still in progress, to be addressed in the next update. The userActions (Issue #106) tests are mostly complete, covering unit and integration tests for core functionalities. However, MainTab and AuthStack (Issues #105 and #104) span across the entire app, so there is still substantial work needed to fully integrate and test all components.

- **Issue #106: UserActions Unit and Integration Testing**
  - **Unit tests**: Completed for all user actions, including authentication, profile updates, and data queries.
  - **Integration tests**: In-progress, with remaining tests focused on user profile updates, user metrics data, and loading user data.

- **Issue #105: Main-Tab Integration Testing**
  - **Completed tests**: HomeScreen navigation to Profile and Settings, and tests for ViewHealthDataScreen.
  - **In-progress**: Further component integration testing for DailyInsights and Calendar interactions.

- **Issue #104: Auth-Stack Integration Testing**
  - **Completed tests**: SignInScreen, SignUpScreen, and navigation from SignIn/SignUp to Home.
  - **In-progress**: Full integration testing for ToSModal and app actions.

### Test Folder Structure & Coverage

```plaintext
- __test__
  - __integration-test__
    - __auth-stack-test__
      - SignInHooks.test.js (in-progress)
    - __auth-component-test__
      - SigninToHome.test.js (completed)
      - SignupToHome.test.js (completed)
    - __auth-navigation-test__
      - SigninToHome.test.js (completed)
      - SignupToHome.test.js (completed)
    - __auth-screen-test__
      - SignInScreen.test.js (completed)
      - SignUpScreen.test.js (in-progress)
    - __main-tab-test__
      - HomeToProfile.test.js (completed)
      - HomeToSettings.test.js (completed)
    - __main-screen-test__
      - SettingsScreen.test.js (completed)
      - ViewInsightsScreen.test.js (completed)
    - userActions.integration.test.js (in-progress)
  - __unit-test__
    - __actions-test__
      - appActions.unit.test.js (completed)
      - userActions.unit.test.js (completed)
    - __components-test__
      - AppHeader.test.js (completed)
    - __reducers-test__
      - toastReducer.unit.test.js (completed)
      - userReducer.unit.test.js (completed)
    - __services-test__
    - __utils-test__
      - localStorage.unit.test.js (in-progress)
  - archives
  - logs
  - App_test.js
  - Profile_test.js
```

### Next Steps
- Complete the remaining integration tests for `userActions`.
- Finalize the remaining test cases for `SignUpScreen.test.js`.
- Continue Auth-Stack and Main-Stack Testing
- Continue Unit-Testing
---

### CODE REVIEW

**Comment by cfiguer055** [comment]
Requires Regression testing (Expo build, Android build, and IOS build)
- Will do it tonight before approval

**Comment by cfiguer055** [comment]
Automated Testing Results: 
```
PASS  __test__/__unit-test__/__actions-test__/appActions.unit.test.js (7.759 s)

Test Suites: 16 passed, 16 total
Tests:       100 passed, 100 total
Snapshots:   5 passed, 5 total
Time:        22.564 s
Ran all test suites.
PS D:\Documents\VisualStudio\smart-clothing-app\SmartClothingApp> 
```

**Comment by cfiguer055** [comment]
**Regression Testing:**
No new issues introduced.
Stable integration with new __test__ folder
Additional Notes:
Testing Document:
[Regression Testing Guideline - PR 108.pdf](https://github.com/user-attachments/files/17337452/Regression.Testing.Guideline.-.PR.108.pdf)


Ready to Launch 🚀

Need Review from one of you
@Persian-of-Interest @harshitp9989

**Review by harshitp9989 — APPROVED** [review]

**Pull request merged**
